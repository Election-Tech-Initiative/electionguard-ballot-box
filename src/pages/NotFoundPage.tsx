import React, { useContext } from 'react'
import { RouteComponentProps } from 'react-router-dom'

import ElectionContext from '../contexts/ballotBoxContext'

import Button from '../components/Button'

const NotFoundPage = (props: RouteComponentProps) => {
  const { reset } = useContext(ElectionContext)
  const { pathname } = props.location
  return (
    <div>
      <h1>Page Not Found.</h1>
      <p>
        No page exists at <code>{pathname}</code>.
      </p>
      <p>
        <Button onPress={() => reset()}>Start Over</Button>
      </p>
    </div>
  )
}

export default NotFoundPage
