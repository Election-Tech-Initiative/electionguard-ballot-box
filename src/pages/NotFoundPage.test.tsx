import React from 'react'
import { Route } from 'react-router-dom'
import { fireEvent } from '@testing-library/react'

import { render } from '../../test/testUtils'

import NotFoundPage from './NotFoundPage'

it('renders NotFoundPage', () => {
  const reset = jest.fn()
  const { container, getByText } = render(
    <Route path="/" component={NotFoundPage} />,
    {
      reset,
      route: '/foobar-not-found-path',
    }
  )
  expect(container.firstChild).toMatchSnapshot()
  fireEvent.click(getByText('Start Over'))
  expect(reset).toHaveBeenCalled()
})
