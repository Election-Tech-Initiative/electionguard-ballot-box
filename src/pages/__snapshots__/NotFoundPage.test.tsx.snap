// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders NotFoundPage 1`] = `
<div>
  <h1>
    Page Not Found.
  </h1>
  <p>
    No page exists at 
    <code>
      /foobar-not-found-path
    </code>
    .
  </p>
  <p>
    <button
      class="sc-bwzfXH gBzjSr"
      type="button"
    >
      Start Over
    </button>
  </p>
</div>
`;
